pyccel/ast/literals.py

```diff
@@ -212,6 +212,15 @@ def __repr__(self):
 
     def __str__(self):
         return str(self.python_value)
+    
+    def __add__(self, o):
+        from .internals import CStringExpression, CMacro
```

Comment:
I did have a problem with circular import so I had to import those classes here.
I don't know if it's the right thing to do or if there is a better solution

Reply:
I can't think of anything. This happens occasionally, it usually means the 2 objects should be defined in the same file, but I'm not sure that I would think to look for `CMacro` or `CStringExpression` in `literals.py`. This workaround is fine as long as it doesn't happen too often

Reply:
I think the better solution is to use the magic method `__radd__` to handle this addition in the `CStringExpression` and `CMacro` classes. See [here](https://blog.finxter.com/python-__rxadd__-magic-method/) for details

Reply:
in `A() + B()` the website said that `__radd__` is used only when `__add__` is not implemented in the class A.
but here I will have an implementation in the `LiteralString` because I want `LiteralString("a") + LiteralString("B")` to return a value of `LiteralString("aB") ` so it will always use the `__add__` of the `LiteralString` class

Reply:
If you scroll down it says:
> Python first tries to call the left object’s `__add__()` method `x.__add__(y)`. But this may fail for two reasons:
> 1. The method `x.__add__()` is not implemented in the first place, or
> 2. The method `x.__add__()` is implemented but returns a NotImplemented value indicating that the data types are incompatible.
>
> If this fails, Python tries to fix it by calling the `y.__radd__()` for reverse addition on the right operand y.